src/pdn/src/PdnGen.tcl

```diff
@@ -640,6 +642,31 @@ proc check_region {region_name} {
   return $region_name
 }
 
+proc check_switch_cells {switch_cells} {
+  foreach switch_cell $switch_cells {
+    if {[[ord::get_db] findMaster $switch_cell] == "NULL"} {
+      utl::warn PDN 178 "Swicth cell $switch_cell not loaded into the database."
+    }
+  }
+
+  return $switch_cells
+}
+
+proc check_unswitched_power {unswitched_power_net_name} {
+  set block [ord::get_db_block]
+
+  if {[set net [$block findNet $unswitched_power_net_name]] == "NULL"} {
+    set net [odb::dbNet_create $block $unswitched_power_net_name]
+    $net setSpecial
+    $net setSigType "POWER"
+  } else {
+    if {[$net getSigType] != "POWER"} {
+      utl::error PDN 177 "Net $unswitched_power_net_name already exists in the design, but is of signal type [$net getSigType]."
```

Comment:
An error might be a little much, a warning is probably okay and then change the type accordingly.